### DISCUSSION TITLE
Add two Yubikey to KeepassXC

### DISCUSSION THREAD

**Discussion by DiegoDAF** [keepassxreboot/keepassxc#6344] [Ideas]
Hi all!!
I'm testing Yubico Yubikeys 5 NFC. In general, we use KeePassXC to save sensible information. KeePassXC accepts yubikeys but just one... Anyone know if it possible to add more than one to a kdbx file?

Thanks a lot!

**Comment by droidmonkey** [comment]
You can only have 1 per database. There is no sense to have 2 or more.

**Reply by DiegoDAF** [reply]
Hi @droidmonkey, sorry no...

The best practices say two, just think what happened with your access in case you lost one yubikey.

**Reply by droidmonkey** [reply]
You make two identical keys

**Reply by DiegoDAF** [reply]
oh! Yes, sure, I will explore this option... Thank you so much!

I found this [in the Docs section](https://keepassxc.org/docs/#faq-yubikey-broken-yubikey), RTFM first, no?:

```
What happens if I break my YubiKey? Can I create backup keys?
    You should always make a copy of the HMAC secret that is stored on the YubiKey and keep it in a secure location. This can be an analog paper copy, but since the YubiKey personalization tool allows you to program a custom secret into the key, you may as well program a second key with the same secret.
     
Can I register multiple YubiKeys with my KeePassXC database?
    You can only use a single secret for encrypting the database. So you can use multiple YubiKeys, but they all have to be programmed with the same secret (see question above).
```

My Idea is more for an environment where we have multiples keys, multiples dbs, etc; and put the same key in all the yubikeys is not the best idea.

**Reply by DiegoDAF** [reply]
Well, for the moment I will continue testing, I still thinking about this, I cannot say that I fully agree with this limitation.
But it is good to know what you think from the development area. 
I really appreciate that!

The explanation to how to setup the same id in slot 2 in two yubikeys is here:
https://support.yubico.com/hc/en-us/articles/360013779759-Using-Your-YubiKey-with-KeePass
Just remember in the second key you must copy & paste the same seed

Thank you so much!

**Reply by Palmermatic** [reply]
Two keys don't work, even if they have the same seed, because the database checks the Yubikey serial number.  There must be support for multiple keys.

**Reply by phoerious** [reply]
There is support for multiple keys. The serial number is only for automatically selecting the correct one if multiple are plugged in. As long as both share the same HMAC secret, they will both work.

**Reply by Palmermatic** [reply]
The database will not use a different key: "Could not find hardway key with serial number xxxxxx"

[Yubico states ](https://support.yubico.com/hc/en-us/articles/360016614880) that the Yubico PAM module "uses the device serial to lookup the expected response, which prevents the module from working unless each YubiKey is registered ahead of time."

**Reply by droidmonkey** [reply]
That only applies after you have opened the database with a specific key. you can open a database with any key that has the right credentials after you lock it.

**Reply by b4st1en** [reply]
> The best practices say two, just think what happened with your access in case you lost one yubikey.

You are right.
And backups are not enough as you have to backup with some other master key.
We could not consider that it works with Yubikey if we cannot set at least 2 differents keys.

But for Professional use I prefer KeePass + KeyManagerUI using PKI tokens..
Meaning you can have a key file to open a same KeePass database with many user certificates
(ok old 2003 interface, but it has been tested on Bug Bounties (EU paid for at least one) unsure about KeePassXC)

**Comment by Palmermatic** [comment]
Got it, thank you!  I should mention this though:

When the database has both a password and a hardware key, and you leave the
password field blank, the resulting unlock error says HMAC mismatch instead
of, say, invalid composite key.

**Reply by droidmonkey** [reply]
We cannot tell the difference between a bad composite key and a corrupt file. Those look the same.

**Comment by fernvenue** [comment]
Actually, use two YubiKeys per KeePassXC database is not that hard, just make sure your two or more YubiKeys has the same challenge response secret. I have an article [Use Multiple YubiKeys for KeePassXC](https://blog.fernvenue.com/archives/use-multiple-yubikeys-for-keepassxc/) about this issue.

**Reply by diego-feito-stori** [reply]
Yep, yu can found the same question with the answer at https://stackoverflow.com/questions/66817721/is-it-possible-to-add-more-than-one-yubikey-to-a-kdbx-file/
